Mage.Sets/src/mage/cards/p/PriestOfTheHauntedEdge.java

```diff
@@ -29,22 +29,22 @@ public final class PriestOfTheHauntedEdge extends CardImpl {
         filter.add(SuperType.SNOW.getPredicate());
     }
 
-    private static final DynamicValue xValue = new MultipliedValue(new PermanentsOnBattlefieldCount(filter), -1);
+    private static final DynamicValue xValue = new SignInversionDynamicValue(new PermanentsOnBattlefieldCount(filter, null));
 
     public PriestOfTheHauntedEdge(UUID ownerId, CardSetInfo setInfo) {
         super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{1}{B}");
 
         this.addSuperType(SuperType.SNOW);
-        this.subtype.add(SubType.ZOMBIE);
-        this.subtype.add(SubType.CLERIC);
+        this.subtype.add(SubType.ZOMBIE, SubType.CLERIC);
         this.power = new MageInt(0);
         this.toughness = new MageInt(4);
 
-        // {T}, Sacrifice Priest of the Haunted Edge: Target creature gets -X/-X until end of turn, where X is the number of snow lands you control. Activate this ability only any time you could cast a sorcery.
+        // {T}, Sacrifice Priest of the Haunted Edge: Target creature gets -X/-X until end of turn,
+        // where X is the number of snow lands you control. Activate only as a sorcery.
         Ability ability = new ActivateAsSorceryActivatedAbility(
-                Zone.BATTLEFIELD, new BoostTargetEffect(
-                xValue, xValue, Duration.EndOfTurn, true
-        ).setText("target creature gets -X/-X until end of turn, where X is the number of snow lands you control"), new TapSourceCost());
+                new BoostTargetEffect(xValue, xValue, Duration.EndOfTurn),
+                new TapSourceCost()
```

Comment:
Shouldn't this be SacrificeSource?

Reply:
The ability has a cost of "{T}, sacrifice {this}". The sacrifice cost is on the line below (and I didn't change it, it's only highlighted because the parentheses moved to the next line)

Reply:
The ability has a cost of "{T}, sacrifice {this}". The sacrifice cost is on the line below (and I didn't change it, it's only highlighted because the parentheses moved to the next line)